### ISSUE TITLE
leadfoot/element returns undefined when .findByClassName fails

### ISSUE THREAD

**Issue by hughryan** [theintern/leadfoot#99]
Bug found on:
Intern 3.4.2
Leadfoot 1.7.0
Node 4.5
Chrome/Firefox/IE

Example Test Case:

```
command.get('http://www.test.com/testpage')
.findByClassName('OuterClassThatExists')
.findByClassName('InnerClassThatDoesNotExist')
.then(function (element) {
console.log(JSON.stringify(element));
})
.catch(function (error) {
console.log('Did not find the element.');
}
```

Given that OuterClassThatExists exists, and InnerClassThatDoesNotExist does not exist, the second findByClassName is expected to throw a 'NoSuchElement' error, that would be caught by the catch statement and print 'Did not find the element' to the console.

Instead, the second findByClassName is returning undefined, and the then clause is being triggered with element = undefined, as seen in the output.

**Comment by LingSpb** [comment]
@jason0x43 is this a bug or new design?

**Comment by jason0x43** [comment]
That's a bug; a failing `findBy` should reject rather than returning an undefined value.

**Comment by LingSpb** [comment]
We are looking forward to get this one fixed, so that we can upgrade to the newer intern/leadfoot version.
Can you check if my PR can fix this issue @jason0x43 ? I'd very much appreciate!

**Comment by jason0x43** [comment]
The PR looks good, I just haven't quite gotten to it yet.

**Comment by hughryan** [comment]
So this is fixed now?

**Comment by jason0x43** [comment]
It is, we just haven't made a new release yet.

**Comment by jason0x43** [comment]
Fixed in https://github.com/theintern/leadfoot/commit/47e6e5bb7cf9ecdedbe9d8f0cc2f67256348609c, will be included in 1.7.1.
